### PR TITLE
Add default WooPayments promotions specs

### LINKED ISSUE #50350: [Enhancement]: Add WooPayments promotions defaults like we have for suggestions
### Describe the solution you'd like

When WooCommerce.com marketplace suggestions are turned off, for WooPayments promotions (shown on the Settings > Payments page) there are no fallback promotions specs as we have for [the payment gateway suggestions](https://github.com/woocommerce/woocommerce/blob/a42ef08face2e28c6b81ceb343d037fab558f64b/plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/DefaultPaymentGateways.php#L13-L13). We should add such defaults.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vladolaru**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

With [the same approach](https://github.com/woocommerce/woocommerce/blob/ee35abf078a39da753cb1c19e717a0f719ea62c0/plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php#L44-L44) we have for overall Payment Gateway Suggestions, we fallback to evaluating default WooPayments promotions when there are no remote promotions, either because marketplace suggestions are turned off or none remained after evaluating the remote specs.

Closes #50350 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JurassicNinja site with WooCommerce on `trunk` and the Jetpack Beta tester plugin (here is a direct [create link](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce=trunk)) 
1. When prompted, skip the WooCommerce Profiler guided setup and set your store country to `US`
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` and make sure Marketplace Suggestions are **enabled**.
1. Go to `WooCommerce > Settings > Payments` and you should see a WooPayments promotion line at the top of the payment methods table:
  
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` and make sure Marketplace Suggestions are **disabled**.
1. Go to `WooCommerce > Settings > Payments` and you should NOT see a WooPayments promotion line at the top of the payment methods table:
  
1. Change the WooCommerce branch to `update/wcpay-promotion-add-default-promotions` from `Jetpack > Beta Tester > WooCommerce > Manage`, search for the branch name and click on Activate:
   
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` and make sure Marketplace Suggestions are **enabled**.
1. Go to `WooCommerce > Settings > Payments` and you should see a WooPayments promotion line at the top of the payment methods table (the same as on the `trunk` branch):
  
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` and make sure Marketplace Suggestions are **disabled**.
1. Go to `WooCommerce > Settings > Payments` and, this time, you should see a WooPayments promotion line at the top of the payment methods table:
  

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/php/src/Internal/Admin/WCPayPromotion/InitTest.php [hunk 1]
```diff
@@ -116,4 +117,15 @@ public function test_matching_suggestions() {
 		$promotions = WCPayPromotion::get_promotions();
 		$this->assertEquals( 'mock-gateway', $promotions[0]->id );
 	}
+
+	/**
+	 * Test that empty suggestions fallback to default ones.
+	 */
+	public function test_fallback_to_default_suggestions() {
+		update_option( 'woocommerce_default_country', 'US' );
```
Comment (ilyasfoo) [inline plugins/woocommerce/tests/php/src/Internal/Admin/WCPayPromotion/InitTest.php:239]: In order to emulate a "default" behaviour, I believe we need to set the `woocommerce_show_marketplace_suggestions` option to `no`: https://github.com/woocommerce/woocommerce/blob/a9f1a812a68a1aa5b5d227d50fafa9620eb604b3/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php#L163

**Review by chihsuan — APPROVED** [review]
PR test instructions are clear and easy to follow. Appreciate the effort you put into this. I've tested this PR and it works as expected. 👍

**Review by ilyasfoo — COMMENTED** [review]
Tests well! Minor nit comment:

File: plugins/woocommerce/tests/php/src/Internal/Admin/WCPayPromotion/InitTest.php (hunk 1 above)
Comment (vladolaru) [reply plugins/woocommerce/tests/php/src/Internal/Admin/WCPayPromotion/InitTest.php:239]: Good call, Ilyas! I took this opportunity to [revamp the unit tests](https://github.com/woocommerce/woocommerce/pull/50345/commits/21e6733e005568ed671fa97afc37eb5977fd1458) as I've very recently done it for [the payment gateway suggestions.](https://github.com/woocommerce/woocommerce/pull/50390)

File: plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php [hunk 2]
```diff
@@ -144,12 +156,21 @@ public static function delete_specs_transient() {
...
 	 */
 	public static function get_specs() {
 		if ( get_option( 'woocommerce_show_marketplace_suggestions', 'yes' ) === 'no' ) {
-			return array();
+			return DefaultPromotions::get_all();
+		}
+
+		$specs = WCPayPromotionDataSourcePoller::get_instance()->get_specs_from_data_sources();
+		// On empty remote specs, fallback to default ones.
+		if ( false === $specs || ! is_array( $specs ) || 0 === count( $specs ) ) {
```
Comment (ilyasfoo) [inline plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php]: Nit: I think `false === $specs` is redundant here, `is_array` check is sufficient for boolean values

**Review by ilyasfoo — APPROVED** [review]
Nice, LGTM! One nit comment, but pre-approving:

File: plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php (hunk 2 above)
Comment (vladolaru) [reply plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php]: Yeah. Definitely overkill. Fixing it.

**Pull request merged**